flake.nix

```diff
@@ -0,0 +1,48 @@
+{
+  description = "Alternative Haskell Infrastructure for Nixpkgs";
+
+  edition = 201909;
+
+  outputs = { self }: {
+    overlay = final: prev:
+      let
+        composeExtensions = f: g: self: super:
+          let
+            fApplied = f self super;
+            super' = super // fApplied;
+          in fApplied // g self super';
+      in (builtins.foldl' composeExtensions (_: _: { }) (import ./overlays))
+      final prev;
+    config = import ./config.nix;
+    sources = import ./nixpkgs;
+    checks = let
```

Comment:
Probably. I missed that somehow :)

Reply:
The issue with `ci.nix` is that we need to explicitly pass `system` to nixpkgs in restricted mode (because `builtins.currentSystem` is not available).

Reply:
Ah, the problem is the generic nixpkgs we use for library functions I guess. We could get rid of that but then we'd have to write our own impls...

Reply:
I'd really prefer not to have all this duplicated logic here, so I'd prefer one of:
- Make `ci.nix` work in the flake
- Just don't have any checks (I can't see anyone running them, tbh)

Reply:
I'm currently working on that. I think we can just `import "${nixpkgs-default}/lib"` without specifying the system?

Reply:
And BTW no, `ci.nix` doesn't implement the same logic. What flake needs as `checks.*` is an attrset of derivations and only derivations, and `R1909.linux.native.tests` is still a mess of stuff. Removed checks for now.

Reply:
Eventually checks will have to be implemented in a flake-friendly way since upstream seems _really_ serious about flakes.

Reply:
I'm not holding my breath :)